### PR TITLE
Feature/microservice architecture templates

### PR DESCRIPTION
**Pull request by HendrikZe**
This introduces functionality behind the menu option "Import Rule Presets".
As an example the Micorservice Architecture has been used to introduce rule/mapping templates so that the user does not need to create the rules/mappings from scratch. The user is guided through a dialog where in the end, after entering a couple of different information on the application that the conformance check should be performed on, rules and mappings are created. 

Note: Wiki pages will be added in the next days.

### CODE REVIEW

File: web-ui/src/main/java/org/archcnl/ui/MainPresenter.java [hunk 1]
```diff
@@ -100,9 +98,6 @@ public MainPresenter() throws ConceptDoesNotExistException {
                 event -> event.handleEvent(conceptManager.getInputConcepts()));
         inputPresenter.addListener(
                 ConceptSelectedEvent.class, event -> event.handleEvent(conceptManager));
-        inputPresenter.addListener(DeleteRuleButtonPressedEvent.class, this::handleEvent);
-        inputPresenter.addListener(DeleteConceptRequestedEvent.class, this::handleEvent);
-        inputPresenter.addListener(DeleteRelationRequestedEvent.class, this::handleEvent);
```
Comment (VincentRaudszus) [inline web-ui/src/main/java/org/archcnl/ui/MainPresenter.java:105]: These three listeners need to be added back. They were recently introduced in [PR 272](https://github.com/Mari-Wie/ArchCNL/pull/272).

File: web-ui/src/main/java/org/archcnl/ui/MainPresenter.java [hunk 2]
```diff
@@ -142,21 +138,6 @@ private void handleEvent(final RelationHierarchySwapRequestedEvent event) {
...
-
-    private void handleEvent(final DeleteConceptRequestedEvent event) {
-        conceptManager.removeConcept(event.getConcept());
-        updateHierarchies(conceptManager, event.getSource());
-    }
-
-    private void handleEvent(final DeleteRelationRequestedEvent event) {
-        relationManager.removeRelation(event.getRelation());
-        updateHierarchies(relationManager, event.getSource());
-    }
```
Comment (VincentRaudszus) [inline web-ui/src/main/java/org/archcnl/ui/MainPresenter.java:158]: Like the listeners the handling methods must also not be removed.

**Review by VincentRaudszus — CHANGES_REQUESTED** [review]
Looks good overall but three listeners need to be added back to the MainPresenter.
Also, importing the _Layered Architecture_ Preset seems to have no effect. Is this intentional?

**Review by VincentRaudszus — APPROVED** [review]

**Comment by HendrikZe** [comment]
@VincentRaudszus Layered Architecture is intentional. However, I agree that this is a bit misleading. I could add some rules for that style as well so that the dialog is behaving as intended (at the moment there are no rules after selecting the style) + preventing the user to be able to create Rules & Mappings (in the last step) by e.g. showing a notification that this style has no templates pre-defined.

Other solution would be to remove the layered architecture and open an issue on this so it can be added in the future. What do you think?

**Comment by VincentRaudszus** [comment]
If you could add some rules and the notification like in the first option that would be perfect. That way it would be clear to every user that further presets will follow in that menu. However, I do not want to ask anything of you that would impact your remaining thesis work.

**Pull request merged**